src/toxencryptsave/encryptsave.rs

```diff
@@ -0,0 +1,165 @@
+/*
+    Copyright © 2016 Zetok Zalbavar <zexavexxe@gmail.com>
+
+    This file is part of Tox.
+
+    Tox is libre software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Tox is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Tox.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+use sodiumoxide::crypto::pwhash::{
+    SALTBYTES, MEMLIMIT_INTERACTIVE, OPSLIMIT_INTERACTIVE,
+    Salt, OpsLimit,
+    gen_salt, derive_key
+};
+use sodiumoxide::crypto::box_::{
+    PRECOMPUTEDKEYBYTES, NONCEBYTES, MACBYTES,
+    Nonce, PrecomputedKey,
+    gen_nonce
+};
+use sodiumoxide::crypto::hash::sha256;
+use ::toxcore::crypto_core;
+
+pub const MAGIC_LENGTH: usize = 8;
+pub const MAGIC_NUMBER: &'static [u8; MAGIC_LENGTH] = b"toxEsave";
+pub const SALT_LENGTH: usize = SALTBYTES;
+pub const KEY_LENGTH: usize = PRECOMPUTEDKEYBYTES;
```

Comment:
In toxcore named TOX_PASS_KEY_LENGTH, I removed the prefix. Maybe you
can be a better name?

在 2016年05月19日 20:30, zetok 写道:
>
> In src/toxencryptsave/encryptsave.rs
> <https://github.com/zetok/tox/pull/9#discussion_r63868971>:
>
> > +    SALTBYTES, MEMLIMIT_INTERACTIVE, OPSLIMIT_INTERACTIVE,
> > +    Salt, OpsLimit,
> > +    gen_salt, derive_key
> > +};
> > +use sodiumoxide::crypto::box_::{
> > +    PRECOMPUTEDKEYBYTES, NONCEBYTES, MACBYTES,
> > +    Nonce, PrecomputedKey,
> > +    gen_nonce
> > +};
> > +use sodiumoxide::crypto::hash::sha256;
> > +use ::toxcore::crypto_core;
> > +
> > +pub const MAGIC_LENGTH: usize = 8;
> > +pub const MAGIC_NUMBER: &'static [u8; MAGIC_LENGTH] = b"toxEsave";
> > +pub const SALT_LENGTH: usize = SALTBYTES;
> > +pub const KEY_LENGTH: usize = PRECOMPUTEDKEYBYTES;
>
> |KEY_LENGTH| on the other hand is ambiguous – just by looking at its
> name you can't really tell what kind of key it is, whether it's
> important, etc.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/zetok/tox/pull/9/files/2500d87ae88c3a989fa1ac9b3d88ce62271f8a67#r63868971>
>